src/common/transformations/src/transformations/op_conversions/convert_fc_to_compressed.cpp

```diff
@@ -0,0 +1,185 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/op_conversions/convert_fc_to_compressed.hpp"
+
+#include <algorithm>
+#include <memory>
+
+#include "openvino/core/rt_info.hpp"
+#include "openvino/core/type/element_type.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/convert.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/op/subtract.hpp"
+#include "openvino/op/transpose.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "openvino/pass/pattern/op/pattern.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "ov_ops/fully_connected.hpp"
+#include "ov_ops/fully_connected_compressed.hpp"
+#include "ov_ops/fully_connected_quantized.hpp"
+#include "ov_ops/placeholder.hpp"
+#include "transformations/utils/utils.hpp"
+
+ov::pass::ConvertFullyConnectedToFullyConnectedCompressed::ConvertFullyConnectedToFullyConnectedCompressed(
+    const std::vector<ov::element::Type>& supported_compression_types,
+    std::function<bool(size_t, size_t, size_t)> supports_config,
+    bool convert_u4zp_to_u8) {
+    using namespace ov::pass::pattern;
+
+    auto compressed_constant = [supported_compression_types](const ov::Output<ov::Node>& output) {
+        return std::any_of(supported_compression_types.begin(),
+                           supported_compression_types.end(),
+                           [&output](const ov::element::Type type) {
+                               return output.get_element_type() == type;
+                           });
+    };
+
+    auto reshape_3d_to_2d = [](const ov::Output<ov::Node>& output) {
+        auto in_ps = output.get_node()->get_input_partial_shape(0);
+        auto out_ps = output.get_node()->get_output_partial_shape(0);
+        return in_ps.rank().is_static() && out_ps.rank().is_static() && in_ps.size() == 3 && out_ps.size() == 2;
+    };
+
+    auto weights_m = wrap_type<ov::op::v0::Constant>(compressed_constant);
```

Comment:
Please use the existing predicate instead of custom one:
```suggestion
    auto weights_m = wrap_type<ov::op::v0::Constant>(ov::pass::pattern::type_matches_any(supported_compression_types));
```